### PR TITLE
Stylistic-updates

### PR DESCRIPTION
**Pull request by sibteali786**
## Describe your changes
- added cursor pointer for Icon 
- added divider in Popupmenu 
- added group selections for Dropdown
## Issue ticket number and link

Closes #

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have added tests
- [ ] I have added a changeset `pnpm changeset add`
- [x] I have added example usage in the kitchen sink app

### CODE REVIEW

File: packages/runtime/src/runtime/customWidget.tsx [hunk 1]
```diff
@@ -19,9 +25,35 @@ export const createCustomWidget = (
     });
     return (
       <CustomScopeProvider value={values ?? inputs}>
-        {EnsembleRuntime.render([widget.body])}
+        <OnLoadAction action={widget?.onLoad} context={{ ...inputs }}>
```
Comment (evshi) [inline packages/runtime/src/runtime/customWidget.tsx:28]: Should be values and then inputs just like line 21

File: packages/runtime/src/runtime/customWidget.tsx [hunk 2]
```diff
@@ -13,6 +13,7 @@ import { useEnsembleAction } from "./hooks/useEnsembleAction";
 
 export interface CustomWidgetProps {
   inputs: Record<string, unknown>;
+  onLoad: EnsembleAction;
```
Comment (evshi) [inline packages/runtime/src/runtime/customWidget.tsx:16]: The props are not needed if it's on the widget model.

File: packages/runtime/src/widgets/Column.tsx [hunk 3]
```diff
@@ -36,7 +36,7 @@ export const Column: React.FC<FlexboxProps> = (props) => {
     if (!action) {
       return;
     }
-    action.callback();
+    action.callback({ ...parentScope });
```
Comment (evshi) [inline packages/runtime/src/widgets/Column.tsx:39]: The scope should be injected by the action, not passed in.

File: packages/runtime/src/widgets/Icon.tsx [hunk 4]
```diff
@@ -30,7 +33,7 @@ export const Icon: React.FC<IconProps> = ({ onTap, ...props }) => {
   return (
     <IconComponent
       className={values?.styles?.names}
-      onClick={(): unknown => onTapActionCallback?.callback()}
+      onClick={(): unknown => onTapActionCallback?.callback({ ...scope })}
```
Comment (evshi) [inline packages/runtime/src/widgets/Icon.tsx:36]: Scope should be injected by the action, not passed in. Which action is missing the scope?

File: packages/runtime/src/widgets/Row.tsx [hunk 5]
```diff
@@ -34,7 +34,7 @@ export const Row: React.FC<FlexboxProps> = (props) => {
     if (!action) {
       return;
     }
-    action.callback();
+    action.callback({ ...parentScope });
```
Comment (evshi) [inline packages/runtime/src/widgets/Row.tsx:37]: Same thing, scope should be injected by the action. The parent scope should get removed here, it's automatically provided in the custom scope provider.

File: packages/runtime/src/widgets/Skeleton.tsx [hunk 6]
```diff
@@ -16,7 +17,7 @@ export const Skeleton: React.FC<SkeletonProps> = (props) => {
       animation={values?.useShimmer ? "wave" : false}
       height={values?.styles?.height}
       style={values?.styles}
-      variant="rectangular"
+      variant={values?.variant ? values?.variant : "rectangular"}
```
Comment (evshi) [inline packages/runtime/src/widgets/Skeleton.tsx:20]: Can use nullish coalescing ??

File: packages/runtime/src/widgets/Icon.tsx (hunk 4 above)
Comment (evshi) [reply packages/runtime/src/widgets/Icon.tsx:36]: It's executeCode that is not injecting the custom scope, not the onTap trigger. This should be fixed on the latest main.

File: packages/runtime/src/widgets/ToggleButton.tsx [hunk 7]
```diff
@@ -30,6 +30,7 @@ interface ToggleButtonPropsStyles {
   borderWidth?: number;
   selectedBorderColor?: Expression<string>;
   selectedBorderWidth?: number;
+  buttonWidth?: Expression<string>;
```
Comment (evshi) [inline packages/runtime/src/widgets/ToggleButton.tsx:33]: What are you using this for? The button widths can derive their width from the overall Toggle Button width

Comment (sibteali786) [reply packages/runtime/src/widgets/ToggleButton.tsx:33]: well it is not, in my case i wanted to apply 50 % width to two toggle buttons

**Review by evshi — APPROVED** [review]

**Pull request merged**